### ISSUE TITLE
The following policy rules failed: [\"ExpirationRule\"]"

### ISSUE THREAD

**Issue by ozbillwang** [netr0m/az-pim-cli#66]
### Welcome

- [X] Yes, I've searched similar [issues on GitHub](https://github.com/netr0m/az-pim-cli/issues) and didn't find any.

### Description of the problem

With latest release, I can list role now. 
```
$ export PIM_TOKEN=xxxx

$ az-pim-cli list role
== Azure Subscrioption 1 ==
	 - Groups Administrator
	 - User Administrator
	 - Application Administrator
	 - Security Reader
```
But when I try to activate the role, I got error `"The following policy rules failed: [\"ExpirationRule\"]"`
```
$ az-pim-cli activate role -n "Azure Subscrioption 1"
2024/10/23 10:09:53 Activating role 'Groups Administrator' for Entra role 'Azure Subscrioption 1' with reason 'config' (ticket:  [])
2024/10/23 10:09:57 The upstream API responded with status 400 Bad Request: {"error":{"code":"RoleAssignmentRequestPolicyValidationFailed","message":"The following policy rules failed: [\"ExpirationRule\"]"}}
```

### Version of az-pim-cli

<details>

```console
$ az-pim-cli version
az-pim-cli version (devel) (built with go1.22.4 from (ec7738f7f35898ed35c47c7f24e643e45042bc80, modified: false, mod sum: "") on 2024-10-21T11:33:18Z)
```

</details>

### Configuration

<details>

```console
export PIM_TOKEN=xxxx
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Output of command

<details>

```console
# paste output here
# make sure to sensor any confidential output
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

**Comment by ozbillwang** [comment]
Could we have an extra parameter "-d" or "--debug" to enable `debug` for any sub-commands when facing issues?

**Comment by netr0m** [comment]
My initial guess is that this resource requires a limited window of activation ([default is 480 min, i.e. 8 hours](https://github.com/netr0m/az-pim-cli/blob/ec7738f7f35898ed35c47c7f24e643e45042bc80/pkg/pim/const.go#L28)). Could you retry with the parameter `--duration <MINUTES>`, e.g. `--duration 30`, and see if that helps?

From the [help message](https://github.com/netr0m/az-pim-cli/blob/main/README.md#entra-roles-1):
```
  -d, --duration int           Duration in minutes that the role should be activated for (default 480)
```

**Comment by netr0m** [comment]
> Could we have an extra parameter "-d" or "--debug" to enable `debug` for any sub-commands when facing issues?

I'm working on an improved logging solution with this goal in mind. Stay tuned!

**Comment by ozbillwang** [comment]
> Try lowering the duration (e.g. --duration 30), and it should be resolved.

Thanks, it works after add `-d 30`. 

And I test with `-d 480`, it shows the error `The following policy rules failed: [\"ExpirationRule\"]"` again. 

After manually check in Azure Console, I found the reason is, in our company, the max Duration is 6 hours ( 360 mins)

**Comment by ozbillwang** [comment]
Another problem with `role`

When I run the command without `-r `, it only activates the first role. 

Is this the design?

If I need activate the rest, I have to run as command 

```
az-pim-cli activate role -n "Azure Subscrioption 1" -r "Application Administrator" -d 60
```

But got a new issue

```
2024/10/23 22:38:01 Activating role 'Application Administrator' for Entra role 'Azure Subscrioption 1' with reason 'config' (ticket:  [])
2024/10/23 22:38:15 The upstream API responded with status 400 Bad Request: {"error":{"code":"RoleAssignmentRequestPolicyValidationFailed","message":"The following policy rules failed: [\"TicketingRule\"]"}}
```

**Comment by ozbillwang** [comment]
I have fixed the issue by add `-T`

```
az-pim-cli activate role -n "Azure Subscrioption 1" -r "Application Administrator" -T "ticket" -d 30
```

**Comment by netr0m** [comment]
> Could we have an extra parameter "-d" or "--debug" to enable `debug` for any sub-commands when facing issues?

Hi @ozbillwang ,

I have now implemented a `--debug` flag to ease the process of troubleshooting (as part of #68). It will is included in release [`v1.4.0`](https://github.com/netr0m/az-pim-cli/releases/tag/v1.4.0)

**Comment by ozbillwang** [comment]
> > Could we have an extra parameter "-d" or "--debug" to enable `debug` for any sub-commands when facing issues?
> 
> Hi @ozbillwang ,
> 
> I have now implemented a `--debug` flag to ease the process of troubleshooting (as part of #68). It is included in release [`v1.4.0`](https://github.com/netr0m/az-pim-cli/releases/tag/v1.4.0)

Thanks, with `--debug` I got more information now.
